### ISSUE TITLE
Not building: error: ‘void* obs_frontend_add_dock(void*)’ is deprecated

### ISSUE THREAD

**Issue by eribertomota** [exeldro/obs-scene-notes-dock#14]
Dear @exeldro.

This plugin is no longer building over Debian with OBS 30.1.2 and GCC 13.2.

Relevant part (hopefully):
```
> /usr/bin/c++ -DHAVE_OBSCONFIG_H -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -Dscene_notes_dock_EXPORTS -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/scene-notes-dock_autogen/include -isystem /usr/include/obs -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Werror -Wextra -Wvla -Wformat -Wformat-security -Wswitch -Wunused-parameter -Wno-unused-function -Wno-missing-field-initializers -fno-strict-aliasing -Wconversion-null -fPIC -mmmx -msse -msse2 -MD -MT CMakeFiles/scene-notes-dock.dir/scene-notes-dock.cpp.o -MF CMakeFiles/scene-notes-dock.dir/scene-notes-dock.cpp.o.d -o CMakeFiles/scene-notes-dock.dir/scene-notes-dock.cpp.o -c /<<PKGBUILDDIR>>/scene-notes-dock.cpp
> /<<PKGBUILDDIR>>/scene-notes-dock.cpp: In function ‘bool obs_module_load()’:
> /<<PKGBUILDDIR>>/scene-notes-dock.cpp:32:30: error: ‘void* obs_frontend_add_dock(void*)’ is deprecated [-Werror=deprecated-declarations]
>    32 |         obs_frontend_add_dock(new SceneNotesDock(main_window));
>       |         ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> In file included from /<<PKGBUILDDIR>>/scene-notes-dock.hpp:3,
>                  from /<<PKGBUILDDIR>>/scene-notes-dock.cpp:1:
> /usr/include/obs/obs-frontend-api.h:142:14: note: declared here
>   142 | EXPORT void *obs_frontend_add_dock(void *dock);
>       |              ^~~~~~~~~~~~~~~~~~~~~
> cc1plus: all warnings being treated as errors
> make[3]: *** [CMakeFiles/scene-notes-dock.dir/build.make:100: CMakeFiles/scene-notes-dock.dir/scene-notes-dock.cpp.o] Error 1
```

The full build log is available from:
http://qa-logs.debian.net/2024/01/15/obs-scene-notes-dock_0.2.0-2_unstable.log

Thanks!

**Comment by exeldro** [comment]
That should be fixed by this commit on master: https://github.com/exeldro/obs-scene-notes-dock/commit/db873c5c0b8b81afc948202d1bf5ae46cd92d125

Do you need a new tag / release for that or is that change not working for you?

**Comment by eribertomota** [comment]
> That should be fixed by this commit on master: [db873c5](https://github.com/exeldro/obs-scene-notes-dock/commit/db873c5c0b8b81afc948202d1bf5ae46cd92d125)

Thanks!

> Do you need a new tag / release for that or is that change not working for you?

This is not needed. I applied a patch[1]. Thanks again!

[1] https://salsa.debian.org/debian/obs-scene-notes-dock/-/blob/debian/master/debian/patches/020_fix-FTBFS.patch
